evaluator/src/main/kotlin/ProjectSourceRule.kt

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (C) 2022 EPAM Systems, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.evaluator
+
+import java.io.File
+
+import org.ossreviewtoolkit.model.OrtResult
+
+/**
+ * An [OrtResultRule] which allows downloading the project's source code if needed.
+ */
+open class ProjectSourceRule(
+    ruleSet: RuleSet,
+    name: String,
+    projectSourceTreeResolver: SourceTreeResolver = ruleSet.ortResult.createProjectSourceTreeResolver()
```

Comment:
What about `sourceTreeResolver` (like the type) or `projectSourceRule` (like the rule class) to make it shorter?